### ISSUE TITLE
Race condition sometimes causes zsh to be used when zsh is disabled

### ISSUE THREAD

**Issue by PaulSD** [hassio-addons/addon-ssh#683]
When this add-on's container is started, the init-ssh and init-user scripts are run in parallel.  When zsh is disabled, if the `chpasswd` command in init-ssh ([here](https://github.com/hassio-addons/addon-ssh/blob/b66f79e1094772ca2acbe4fcab97fea84a15a821/ssh/rootfs/etc/s6-overlay/s6-rc.d/init-ssh/run#L121)) runs at the same time as the `sed` command in init-user that changes `zsh` to `bash` in `/etc/passwd` ([here](https://github.com/hassio-addons/addon-ssh/blob/b66f79e1094772ca2acbe4fcab97fea84a15a821/ssh/rootfs/etc/s6-overlay/s6-rc.d/init-user/run#L55)) then the `chpasswd` command may read `/etc/passwd` before `sed` writes it, then write `/etc/passwd` after `sed` writes it, effectively negating the `sed` command and reverting shell back to `zsh` in `/etc/passwd`.  This seems to happen to me fairly regularly (more than 50% of the times that I start/restart this add-on).

I can think of a number of ways to fix this, but I'm not confident that I know enough to choose the "best" fix:
* The comment in the init-ssh script mentions "We need to set a password for the user account", although I'm not aware of any reason the user account should actually need a password.  Maybe we could simply skip `chpasswd` if the user hasn't configured a password?  (Although this won't fix the problem if the user has both disabled zsh and configured a password.)
* Instead of changing the shell in `/etc/passwd`, we could build the container with the shell set to something like `/usr/local/bin/shell` and then (in init-user) make `/usr/local/bin/shell` a symlink to either zsh or bash as appropriate.
* We could add a s6 dependency between init-ssh and init-user to avoid running those scripts in parallel.
* `chpasswd` has a locking mechanism for /etc/passwd ([here](https://github.com/shadow-maint/shadow/blob/9340efbb0d60ae73222b6d049786725397043802/lib/commonio.c#L357)).  We could add code to the init-user script to implement this locking mechanism around the `sed` command, to prevent parallel modifications to `/etc/passwd`.

**Comment by frenck** [comment]
Nice catch, we can just add a dependency here 👍
